### PR TITLE
Conditionalize fuzz tests on feature macros

### PR DESCRIPTION
**Pull request by kaduk**
Do not try to fuzz-test structures/routines that are compiled
out of the library due to library configuration.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: fuzz/asn1.c [hunk 1]
```diff
@@ -319,17 +320,24 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
...
     DO_TEST(DH, d2i_DHparams, i2d_DHparams, DHparams_print);
     DO_TEST(DH, d2i_DHxparams, i2d_DHxparams, DHparams_print);
+#endif
+#ifndef OPENSSL_NO_DSA
     DO_TEST_NO_PRINT(DSA_SIG, d2i_DSA_SIG, i2d_DSA_SIG);
     DO_TEST_PRINT_OFFSET(DSA, d2i_DSAPrivateKey, i2d_DSAPrivateKey, DSA_print);
     DO_TEST_PRINT_OFFSET(DSA, d2i_DSAPublicKey, i2d_DSAPublicKey, DSA_print);
     DO_TEST(DSA, d2i_DSAparams, i2d_DSAparams, DSAparams_print);
+#endif
     DO_TEST_PRINT_OFFSET(RSA, d2i_RSAPublicKey, i2d_RSAPublicKey, RSA_print);
```
Comment (kroeckx) [inline fuzz/asn1.c]: We seem to have an OPENSSL_NO_RSA ...

Comment (kroeckx) [reply fuzz/asn1.c]: But no config option to actually do it

Comment (kaduk) [reply fuzz/asn1.c]: Right, it's not in ```%disablables```, so you get an error if you try to configure with no-rsa.

**Comment by kaduk** [comment]
pushed; closing.

**Pull request closed without merging**